### PR TITLE
Add verify job for community repo

### PR DESCRIPTION
**Pull request by cblecker**
I haven't added a new CI job before, so I'm not sure if this is all the pieces required, or if there are more.

Ref: #3034

### CODE REVIEW

**Comment by BenTheElder** [comment]
What all does `make verify` do exactly? Can we add this as a Prow job instead? https://github.com/kubernetes/test-infra/issues/3778

**Comment by cblecker** [comment]
It creates a docker image containing a go application, runs it, and then diffs against what's in the repo.

https://github.com/kubernetes/community/blob/master/hack/verify.sh

**Comment by BenTheElder** [comment]
> It creates a docker image containing a go application, runs it, and then diffs against what's in the repo.

Can we just do this without docker then? We're looking at DinD support in one of the Prow images but in the meantime we plan to turn down Jenkins by ~ year end and don't support running docker commands on Prow. cc @krzyzacy @fejta

**Comment by cblecker** [comment]
Yeah, dockerizing this one little go app is a little complex for what is needed, I think. Let me look into it.

**Comment by BenTheElder** [comment]
> Yeah, dockerizing this one little go app is a little complex for what is needed, I think. Let me look into it.

Thank you! :smile:

**Comment by cblecker** [comment]
@BenTheElder PTAL and see if I have the right format for a prow job.

/hold
https://github.com/kubernetes/community/pull/1267 needs to merge before this can :)

File: prow/config.yaml [hunk 1]
```diff
@@ -42,6 +42,42 @@ presubmits:
...
+          value: /etc/service-account/service-account.json
+        volumeMounts:
+        - name: service
+          mountPath: /etc/service-account
+          readOnly: true
+        - name: cache-ssd
+          mountPath: /root/.cache
+        ports:
+        - containerPort: 9999
+          hostPort: 9999
```
Comment (BenTheElder) [inline prow/config.yaml]: @krzyzacy what did we decide about the port / cache for non-bazel jobs?

Comment (BenTheElder) [reply prow/config.yaml]: And if we want the cache on this job then probably also `args: - "--git-cache=/root/.cache/git"`

**Comment by BenTheElder** [comment]
I think this looks just fine, I want @krzyzacy to take a look though 😃 
Also you might want to add it to test-grid somewhere? (I'm not sure if that's really necessary but it might make sense).

File: prow/config.yaml (hunk 1 above)
Comment (krzyzacy) [reply prow/config.yaml]: > what did we decide about the port / cache for non-bazel jobs

@spxtr @rmmh shall we just make cache can be shared with multiple jobs? Or that's already the case?

**Comment by krzyzacy** [comment]
@cblecker thanks for doing this :-)
/lgtm
feel free to `/hold cancel` when dependency PR is merged

**Comment by cblecker** [comment]
Interesting that LGTM didn't remove after I pushed a new commit..

**Comment by cblecker** [comment]
Oh.. that's because it's a munger: https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/lgtm_after_commit.go

**Pull request merged**